### PR TITLE
Update base-challenge-template 

### PR DESCRIPTION
**Pull request by Nadai2010**
# Update to base-challenge template 

This version is updated with the latest features from v3 and some improvements integrated into the OD Hack up to PR-316 of Scaffold Stark
## Types of change

- [x] Feature
- [x] Enhancement

### CODE REVIEW

File: packages/nextjs/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
Comment (gianalarcon) [inline packages/nextjs/README.md:1]: Not sure if we really need this file @Nadai2010 @metalboyrick . Can we remove even from scaffold-stark?

File: packages/nextjs/components/assets/BuidlGuidlLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const BuidlGuidlLogo = ({ className }: { className: string }) => {
```
Comment (gianalarcon) [inline packages/nextjs/components/assets/BuidlGuidlLogo.tsx:1]: Check if we really need this asset

**Comment by gianalarcon** [comment]
We have same asset twice
[debug1](https://github.com/Scaffold-Stark/speedrunstark/pull/153/files#diff-8438634413f248b67883d6ea8ea383047db23baf367b054caca70045cd8edf71), [debug2](https://github.com/Scaffold-Stark/speedrunstark/pull/153/files#diff-a245f377181c15689386f73f29be46d3bc750ac4dd79f7664ea3475a8d921b3c)

Let's double check the assets. We don't need all of them to support the challenges. Lest try to make the base template lighter.

File: packages/nextjs/README.md (hunk 1 above)
Comment (metalboyrick) [reply packages/nextjs/README.md]: ya can remove

**Comment by metalboyrick** [comment]
@Nadai2010 @gianalarcon we might need to do this cleanup on the root scaffold as well

**Review by metalboyrick — COMMENTED** [review]
image

Wondering where do these images appear? I think we can remove? I didnt see them in the readme as well

**Review by metalboyrick — APPROVED** [review]
LGTM

File: packages/nextjs/app/debug/_components/contract/ContractReadMethods.tsx [hunk 3]
```diff
@@ -1,7 +1,10 @@
 import { Abi } from "abi-wan-kanabi";
 import {
+  AbiFunction,
```
Comment (gianalarcon) [inline packages/nextjs/app/debug/_components/contract/ContractReadMethods.tsx:3]: Do we really need this libraries? The challenges will need them right?  @Nadai2010 
image

Comment (Nadai2010) [reply packages/nextjs/app/debug/_components/contract/ContractReadMethods.tsx]: Perfect, we will do the same in Scaffold main, I remove them from here now

**Pull request merged**